robottelo/ui/discoveredhosts.py

```diff
@@ -81,7 +82,11 @@ def waitfordiscoveredhost(self, hostname):
 
     def fetch_fact_value(self, hostname, element):
         """Fetch the value of selected fact from discovered hosts page"""
-        host = self.search(hostname)
+        try:
+            host = self.search(hostname)
+        except StaleElementReferenceException:
+            # Repeat action in case of DOM was changed
+            host = self.search(hostname)
```

Comment:
Just a suggestion - maybe it's time to introduce some custom context manager for such workarounds to look not that ugly?
Imo smth like that looks way better:
```python
with retry(exceptions=StaleElementReferenceException, tries=3):
    host = self.search(hostname)
```
Your thoughts, @ldjebran @oshtaier @odovzhenko ?

Reply:
One `try` is enough as DOM changed only once.

Reply:
@odovzhenko @ldjebran you didn't get my point, i wasn't insisting on 3 tries but proposed a common approach to handle such issues. Default `tries` would be set to `2` (retry once), so this specific scenario would be even nicer:
```python
with retry(StaleElementReferenceException):
     host = self.search(hostname)
```

Reply:
I get your point, but I'd rather agree with @oshtaier that we should implement such handles for basic operations that will require whole retesting. =)